src/databricks/labs/remorph/resources/reconcile/queries/installation/aggregate_metrics.sql

```diff
@@ -1,11 +1,11 @@
 CREATE TABLE IF NOT EXISTS aggregate_metrics (
     recon_table_id BIGINT NOT NULL,
-    rule_id BIGINT NOT NULL,
+    rule_id BIGINT,
```

Comment:
When can rule_id be null?

Reply:
There could be cases where provided column names are not present in the table, In those situations we are capturing the exception in the Delta Tables.

Example below:  
`Runtime exception occurred while fetching data using SELECT min(p_discount) AS source_min_p_discount, max(p_price11) AS source_max_p_price11, count(p_id) AS source_count_p_id FROM hive_metastore.default.product_ctgry : [UNRESOLVED_COLUMN.WITH_SUGGESTION] A column, variable, or function parameter with name **p_price11** cannot be resolved. Did you mean one of the following? `

Reply:
reverted `rule_id` to NOT NULL, as per the discussion.